lib/galaxy/security/validate_user_input.py

```diff
@@ -54,32 +63,46 @@ def validate_publicname_str(publicname):
     return ''
 
 
-def validate_email(trans, email, user=None, check_dup=True, allow_empty=False):
+def validate_email(trans, email, user=None, check_dup=True, allow_empty=False, validate_domain=False):
     """
     Validates the email format, also checks whether the domain is blocklisted in the disposable domains configuration.
     """
     if (user and user.email == email) or (email == "" and allow_empty):
         return ''
     message = validate_email_str(email)
-    if message:
-        pass
-    elif check_dup and trans.sa_session.query(trans.app.model.User).filter(func.lower(trans.app.model.User.table.c.email) == email.lower()).first():
-        message = f"User with email '{email}' already exists."
-    #  If the allowlist is not empty filter out any domain not in the list and ignore blocklist.
-    elif trans.app.config.email_domain_allowlist_content is not None:
+    if not message and validate_domain:
         domain = extract_domain(email)
-        if domain not in trans.app.config.email_domain_allowlist_content:
-            message = "Please enter an allowed domain email address for this server."
-    #  If the blocklist is not empty filter out the disposable domains.
-    elif trans.app.config.email_domain_blocklist_content is not None:
-        domain = extract_domain(email, base_only=True)
-        if domain in trans.app.config.email_domain_blocklist_content:
-            message = "Please enter your permanent email address."
+        message = validate_domain(domain)
+
+    if not message and check_dup and trans.sa_session.query(trans.app.model.User).filter(func.lower(trans.app.model.User.table.c.email) == email.lower()).first():
+        message = f"User with email '{email}' already exists."
+
+    if not message:
+        # If the allowlist is not empty filter out any domain not in the list and ignore blocklist.
+        if trans.app.config.email_domain_allowlist_content is not None:
+            domain = extract_domain(email)
+            if domain not in trans.app.config.email_domain_allowlist_content:
+                message = "Please enter an allowed domain email address for this server."
+        # If the blocklist is not empty filter out the disposable domains.
+        elif trans.app.config.email_domain_blocklist_content is not None:
+            domain = extract_domain(email, base_only=True)
+            if domain in trans.app.config.email_domain_blocklist_content:
+                message = "Please enter your permanent email address."
+
+    return message
+
+
+def validate_domain(domain):
+    message = None
+    try:
+        socket.gethostbyname(domain)
+    except socket.gaierror:
+        message = "The email domain cannot be resolved."
```

Comment:
I think we need a very short timeout here, and we might need to think about Galaxy instances that don't have access to the internet ?

Reply:
Good call on the timeout, I'll add that. This feature is only used if email activation is enabled, so it should be safe on inacessible galaxies (which would not enable activation). That said, I can add a separate config option for it if that would be preferable.

Reply:
I'm not sure how to add a timeout here without implementing our own timeout function. `gethostbyname()` is a socket module function, so `settimeout()` won't affect its behavior. Same goes for `setdefaulttimeout()` which affects new socket objects. If there's a way to do it, or if we have a utility function in Galaxy, please let me know and I'll add it; otherwise, if we do need a timeout here, I think it belongs in a separate PR, in which case I'll add an issue.